core/src/parse.rs

```diff
@@ -92,32 +75,26 @@ impl<'a> Parse<'a> for LabelDecl {
 
 impl<'a> Parse<'a> for StackRef {
     fn parse(input: RawSpan<'a>) -> ParseResult<'a, Self> {
-        context(
-            "Stack",
-            map(preceded(tag_no_case("S"), StackId::parse), StackRef),
-        )(input)
+        map(preceded(tag_no_case("S"), StackId::parse), StackRef)(input)
     }
 }
 
 impl<'a> Parse<'a> for RegisterRef {
     fn parse(input: RawSpan<'a>) -> ParseResult<'a, Self> {
-        context(
-            "Register",
-            alt((
-                // "RLI" => RegisterRef::InputLength
-                map(tag_no_case("RLI"), |_| RegisterRef::InputLength),
-                // "RSx" => RegisterRef::StackLength(x)
-                map(
-                    preceded(tag_no_case("RS"), cut(StackId::parse)),
-                    RegisterRef::StackLength,
-                ),
-                // "RXx" => RegisterRef::User(x)
-                map(
-                    preceded(tag_no_case("RX"), cut(UserRegisterId::parse)),
-                    RegisterRef::User,
-                ),
-            )),
-        )(input)
+        alt((
+            // "RLI" => RegisterRef::InputLength
+            map(tag_no_case("RLI"), |_| RegisterRef::InputLength),
+            // "RSx" => RegisterRef::StackLength(x)
```

Comment:
Don't need to address this in this PR but if they did `RL` or some bad tag im not sure what kind of error they would get